guides/howtos/Multi tenancy with query prefixes.md

```diff
@@ -0,0 +1,253 @@
+
+So far we have explored how to set a global prefix using Postgres' and how to set the prefix at the query or struct level. When the global prefix is set, it also changes the prefix migrations run on. However it is also possible to set the prefix through the command line or per table in the migration itself.
+
+For example, imagine you are a gaming company where the game is broken in 128 partitions, named "prefix_1", "prefix_2", "prefix_3" up to "prefix_128". Now, whenever you need to migrate data, you need to migrate data on all different 128 prefixes. There are two ways of achieve that.
+
+The first mechanism is to invoke `mix ecto.migrate` multiple times, once per prefix, passing the `--prefix` option:
+
+```bash
+$ mix ecto.migrate --prefix "prefix_1"
+$ mix ecto.migrate --prefix "prefix_2"
+$ mix ecto.migrate --prefix "prefix_3"
+...
+$ mix ecto.migrate --prefix "prefix_128"
+```
+
+The other approach is by changing each desired migration to run across multiple prefixes. For example:
+
+```elixir
+defmodule MyApp.Repo.Migrations.CreateSample do
+  use Ecto.Migration
+
+  def change do
+    for i <- 1..128 do
+      prefix = "prefix_#{i}"
+      create table(:samples, prefix: prefix) do
+        add :name, :string
+        timestamps()
+      end
+
+      # Execute the commands on the current prefix
+      # before moving on to the next prefix
+      flush()
+    end
+  end
+end
+```
+
+## Schema prefixes
+
+Finally, Ecto allows you to set a particular schema to run on a specific prefix. Imagine you are building a multi-tenant application. Each client data belongs to a particular prefix, such as "client_foo", "client_bar" and so forth. Yet your application may still rely on a set of tables that is shared across all clients. One of such tables may be exactly the table that maps the Client ID to its database prefix. Let's assume we want to store this data in a prefix named "main":
+
+```elixir
+defmodule MyApp.Mapping do
+  use Ecto.Schema
+
+  @schema_prefix "main"
+  schema "mappings" do
+    field :client_id, :integer
+    field :db_prefix
+    timestamps
+  end
+end
+```
+
+Now running `MyApp.Repo.all MyApp.Mapping` will by default run on the "main" prefix, regardless of the value configured globally on the `:after_connect` callback. Similar will happen to `insert`, `update`, and similar operations, the `@schema_prefix` is used unless the `:prefix` is explicitly changed via `Ecto.put_meta/2` or by passing the `:prefix` option to the repository operation.
+
+Keep in mind, however, that queries run on a single prefix. For example, if `MyApp.Mapping` on prefix "main" depends on a schema named `MyApp.Other` on prefix "another", a query starting with `MyApp.Mapping` will always run on the "main" prefix. By design it is not possible to perform query joins across prefixes. If data belongs to different prefixes, it is best to not couple them structurally nor via queries, in order to **keep data in different prefixes isolated**.
```

Comment:
This is no longer true. Ecto 3.0 allows you to do this:

```elixir
from p in Post, prefix: "foo",
  join: c in Comment, prefix: "bar"
```

We need to update this chapter to add a new section. Here are the docs: https://hexdocs.pm/ecto/Ecto.Query.html#module-query-prefix